File: leetcode_js/src/intersectionOfTwoArrays.js
const intersection = function (nums1, nums2) {
  const result = new Set();
  const set = new Set();

  nums1.forEach((n) => set.add(n));
  nums2.forEach((n) => {
    if (set.has(n)) {
      result.add(n);
    }
  });

  return Array.from(result);
};

module.exports = { intersection };

File: leetcode_python/lib/problems/p0349_intersection_of_two_arrays/intersection_of_two_arrays.py
from typing import List


class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        temp = set()
        for i in nums1:
            temp.add(i)

        result = set()
        for i in nums2:
            if i in temp:
                result.add(i)

        return list(result)
